File: C# OOP 22'/C# OOP/Inheritance - Exercise/NeedForSpeed/Vehicle.cs
using System;
using System.Reflection.Metadata.Ecma335;

namespace NeedForSpeed
{
    public class Vehicle
    {
        private const double DEFAULT_FUEL_CONSUMPTION = 1.25;

        public Vehicle(int horsePower, double fuel)
        {
            HorsePower = horsePower;
            Fuel = fuel;
        }

        public double Fuel { get; set; }
        public int HorsePower { get; set; }
        public int MyProperty { get; set; }
        public virtual double FuelConsumption => DEFAULT_FUEL_CONSUMPTION;

        public void Drive(double kilometers)
        {
            double fuelLeft = Fuel - FuelConsumption * kilometers;

            if (fuelLeft >= 0)
                Fuel -= FuelConsumption * kilometers;
        }
    }
}

File: Python/OOP/Inheritance/need_for_speed/vehicle.py
class Vehicle:
    DEFAULT_FUEL_CONSUMPTION = 1.25

    def __init__(self, fuel: float, horse_power: int):
        self.fuel = fuel
        self.horse_power = horse_power
        self.fuel_consumption = Vehicle.DEFAULT_FUEL_CONSUMPTION

    def drive(self, distance):
        needed_fuel = distance * self.fuel_consumption
        if needed_fuel <= self.fuel:
            self.fuel -= needed_fuel